pkg/secretutils/secret_manager.go

```diff
@@ -78,30 +78,24 @@ func (sm *SecretManager) claimSecret(secret *corev1.Secret, owner client.Object,
 			"ownerKind", owner.GetObjectKind().GroupVersionKind().Kind,
 			"owner", owner.GetNamespace()+"/"+owner.GetName(),
 			"ownerUID", owner.GetUID())
-		if ownerIsController {
-			if !metav1.IsControlledBy(secret, owner) {
-				ownerLog.Info("setting secret controller reference")
-				if err := controllerutil.SetControllerReference(owner, secret, sm.client.Scheme()); err != nil {
-					return errors.Wrap(err, "failed to set secret controller reference")
-				}
-				needsUpdate = true
+		alreadyOwned := false
+		ownerUID := owner.GetUID()
+		for _, ref := range secret.GetOwnerReferences() {
+			// We used to add controller references to BMC
+			// secrets. This was wrong, update.
+			if ref.UID == ownerUID && ref.Controller == nil {
+				alreadyOwned = true
+				break
+			} else if ref.Controller != nil && *ref.Controller {
+				ownerLog.Info("updating secret to no longer have an owner of type controller")
```

Comment:
teeechnically this is only true if `ref.UID == owner.UID`, but in practice it seems unlikely that anyone else will add an owner reference to the same Secret.